### ISSUE TITLE
flatten neural net output

### ISSUE THREAD

**Issue by wrongu** [Rochester-NRT/AlphaGo#90]
The network as created by `CNNPolicy.create_network` has outputs that are 2-dimensional (19x19). This messes with the softmax (each row is normalized, outputs sum to 19), and hence with training and with Keras' calculations of loss and accuracy

Need to flatten the output and change other functions accordingly
